packages/camera/camera/android/src/test/java/io/flutter/plugins/camera/CameraTest.java

```diff
@@ -0,0 +1,733 @@
+
+    when(mockExposureOffsetFeature.getExposureOffsetStepSize()).thenReturn(stepSize);
+
+    double actualSize = camera.getExposureOffsetStepSize();
+
+    verify(mockExposureOffsetFeature, times(1)).getExposureOffsetStepSize();
+    assertEquals(stepSize, actualSize, 0);
+  }
+
+  @Test
+  public void getMaxExposureOffset() {
+    ExposureOffsetFeature mockExposureOffsetFeature =
+        mockCameraFeatureFactory.createExposureOffsetFeature(mockCameraProperties);
+    double expectedMaxOffset = 42.0;
+
+    when(mockExposureOffsetFeature.getMaxExposureOffset()).thenReturn(expectedMaxOffset);
+
+    double actualMaxOffset = camera.getMaxExposureOffset();
+
+    verify(mockExposureOffsetFeature, times(1)).getMaxExposureOffset();
+    assertEquals(expectedMaxOffset, actualMaxOffset, 0);
+  }
+
+  @Test
+  public void getMinExposureOffset() {
+    ExposureOffsetFeature mockExposureOffsetFeature =
+        mockCameraFeatureFactory.createExposureOffsetFeature(mockCameraProperties);
+    double expectedMinOffset = 21.5;
+
+    when(mockExposureOffsetFeature.getMinExposureOffset()).thenReturn(21.5);
+
+    double actualMinOffset = camera.getMinExposureOffset();
+
+    verify(mockExposureOffsetFeature, times(1)).getMinExposureOffset();
+    assertEquals(expectedMinOffset, actualMinOffset, 0);
+  }
+
+  @Test
+  public void getMaxZoomLevel() {
+    ZoomLevelFeature mockZoomLevelFeature =
+        mockCameraFeatureFactory.createZoomLevelFeature(mockCameraProperties);
+    float expectedMaxZoomLevel = 4.2f;
+
+    when(mockZoomLevelFeature.getMaximumZoomLevel()).thenReturn(expectedMaxZoomLevel);
+
+    float actualMaxZoomLevel = camera.getMaxZoomLevel();
+
+    verify(mockZoomLevelFeature, times(1)).getMaximumZoomLevel();
+    assertEquals(expectedMaxZoomLevel, actualMaxZoomLevel, 0);
+  }
+
+  @Test
+  public void getMinZoomLevel() {
+    ZoomLevelFeature mockZoomLevelFeature =
+        mockCameraFeatureFactory.createZoomLevelFeature(mockCameraProperties);
+    float expectedMinZoomLevel = 4.2f;
+
+    when(mockZoomLevelFeature.getMinimumZoomLevel()).thenReturn(expectedMinZoomLevel);
+
+    float actualMinZoomLevel = camera.getMinZoomLevel();
+
+    verify(mockZoomLevelFeature, times(1)).getMinimumZoomLevel();
+    assertEquals(expectedMinZoomLevel, actualMinZoomLevel, 0);
+  }
+
+  @Test
+  public void getRecordingProfile() {
+    ResolutionFeature mockResolutionFeature =
+        mockCameraFeatureFactory.createResolutionFeature(mockCameraProperties, null, null);
+    CamcorderProfile mockCamcorderProfile = mock(CamcorderProfile.class);
+
+    when(mockResolutionFeature.getRecordingProfile()).thenReturn(mockCamcorderProfile);
+
+    CamcorderProfile actualRecordingProfile = camera.getRecordingProfile();
+
+    verify(mockResolutionFeature, times(1)).getRecordingProfile();
+    assertEquals(mockCamcorderProfile, actualRecordingProfile);
+  }
+
+  @Test
+  public void setExposureMode_Should_update_exposure_lock_feature_and_update_builder() {
```

Comment:
nit: in Java camel case is used. In tests, underscore is used to workaround the fact Java doesn't have nested groups. Also, if possible break this into two: e.g.  `setExposureMode_shouldUpdateExposureLockFeatures`, `setExposureMode_shouldUpdateBuilder`

Reply:
I think these have been written like this to maintain consistency with the existing tests. 
I've updated the casing on all tests now, as well as split up each `*AndUpdateBuilder` test.